### PR TITLE
Enable more tslint strict rules

### PR DESCRIPTION
**Pull request by vogre**
Discovered when writing an async function and forgetting to await

Also add await-promise rule

### CODE REVIEW

File: common/tslint.yml [hunk 1]
```diff
@@ -3,6 +3,16 @@ extends:
...
+    await-promise: true
+    no-floating-promises: true
+    strict-boolean-expressions:
+      - true
+      - allow-mix
+      - allow-undefined-union
+      - allow-null-union
+      - allow-string
+      - allow-number
+      - ignore-rhs
```
Comment (arielshaqed) [inline common/tslint.yml]: Cool!  (Also good to see TSLint invented a new way to write a very limited set of generics, https://palantir.github.io/tslint/rules/strict-boolean-expressions/)

File: shift-react-app/src/shiftReactApp.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {
   ignoreShift,
 } from './steps';
 
-const steps = [
+const steps: Array<(() => Promise<string>) | (() => string) | (() => void)> = [
```
Comment (arielshaqed) [inline shift-react-app/src/shiftReactApp.ts]: Why is this suddenly required?

File: tools/rush-update/src/types.d.ts [hunk 3]
```diff
@@ -5,5 +5,5 @@ declare module 'npm-check-updates' {
     upgrade?: boolean,
   }
 
-  function run(params: NCUParams): Record<string, string[]>;
+  function run(params: NCUParams): Promise<Record<string, string[]>>;
```
Comment (arielshaqed) [inline tools/rush-update/src/types.d.ts]: This was a bug in the old type definitions, right?

Comment (vogre) [reply tools/rush-update/src/types.d.ts]: Yep! The implementation correctly awaited.

File: shift-react-app/src/shiftReactApp.ts (hunk 2 above)
Comment (vogre) [reply shift-react-app/src/shiftReactApp.ts]: It looks like typescript inferred that the array is of functions of the form `() => void` based on the first function in the array, and allows assigning `() => string` to a `() => void`.

**Pull request merged**